Source/SwiftLintFramework/Rules/OpeningBraceRule.swift

```diff
@@ -29,12 +43,68 @@ public struct OpeningBraceRule: Rule {
     )
 
     public func validateFile(file: File) -> [StyleViolation] {
-        let pattern = "((?:[^( ]|[\\t\\n\\f\\r (] )\\{)"
-        let excludingKinds = SyntaxKind.commentAndStringKinds()
-
-        return file.matchPattern(pattern, excludingSyntaxKinds: excludingKinds).map {
+        return file.violatingOpeningBraceRanges().map {
             StyleViolation(ruleDescription: self.dynamicType.description,
                 location: Location(file: file, offset: $0.location))
         }
     }
+
+    public func correctFile(file: File) -> [Correction] {
+        let fileRegions = file.regions()
+        let violatingRanges = file.violatingOpeningBraceRanges().filter { range in
+            let region = fileRegions.filter {
+                $0.contains(Location(file: file, offset: range.location))
+            }.first
+            return region?.isRuleEnabled(self) ?? true
+        }
+
+        let adjustedRanges = writeToFile(file, violatingRanges: violatingRanges)
+
+        return adjustedRanges.map {
+            Correction(ruleDescription: self.dynamicType.description,
+                location: Location(file: file, offset: $0.location))
+        }
+    }
+
+    private func writeToFile(file: File, violatingRanges: [NSRange]) -> [NSRange] {
+        let correctStringCount = correctString.characters.count
+        var correctedContents = file.contents
+        var adjustedRanges = [NSRange]()
+        var previousLengthDelta = 0
+
+        for violatingRange in violatingRanges {
+            guard let range = file.contents.nsrangeToIndexRange(violatingRange) else {
+                continue
+            }
+            let capturedString = file.contents[range]
```

Comment:
I think this could be simplified quite a bit:

```swift
let capturedString = file.contents[range]

// if "struct Command{" is violated with violating string = "d{",
// adjust range to only replace "{"
let shouldAdjust = capturedString.characters.count == 2 &&
    capturedString.rangeOfCharacterFromSet(whitespaceAndNewlineCharacterSet) == nil
let adjustment = Int(shouldAdjust)

previousLengthDelta = violatingRange.length - correctStringCount - adjustment
let adjustedRange = NSRange(
    location: violatingRange.location - previousLengthDelta + adjustment,
    length: correctStringCount - adjustment
)
adjustedRanges += [adjustedRange]
```